contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -1954,64 +1977,85 @@ class tsqlBuilder : public tsqlCommonMutator
-				Assert(str.front() == '\'');
-				Assert(str.back() == '\'');		    						
-			}
+		if (str.front() == '"')
+		{	
+			Assert(str.back() == '"');
+			
+			// Change double-quoted string to single-quoted string:
+			str = rewriteDoubleQuotedString(str);	
+			size_t startPosition = ctx->start->getStartIndex();
+			if (in_execute_body_batch_parameter) startPosition += fragment_EXEC_prefix.length(); // add length of prefix prepended internally for execute_body_batch  
+			rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair(::getFullText(ctx), str)));
 		}
+		else
+		{
+			// This is a single-quoted string, no further action needed
+			Assert(str.front() == '\'');
+			Assert(str.back() == '\'');		    						
+		}
+	}
+
+	void enterExecute_parameter(TSqlParser::Execute_parameterContext *ctx) override
+	{
+		if (in_execute_body_batch) in_execute_body_batch_parameter = true;
 	}
 
 	void exitExecute_parameter(TSqlParser::Execute_parameterContext *ctx) override
 	{
 		if (ctx->id())
 		{
 			// A stored procedure parameter which is parsed as a column name (= identifier)
-			// is either an unquoted string, or a double-quoted string.
+			// is either an unquoted string, a double-quoted string, or a bracketed string.
 			// For a procedure call parameter, a double-quoted string is interpreted 
-			// as a string even with QUOTED_IDENTIFIER=ON.
-			
+			// as a string even with QUOTED_IDENTIFIER=ON.			
 			std::string str = getFullText(ctx->id());
+			size_t startPosition = ctx->id()->start->getStartIndex();		
+			if (in_execute_body_batch_parameter) startPosition += fragment_EXEC_prefix.length(); // add length of prefix prepended internally for execute_body_batch			
 
 			if (str.front() == '"')
 			{
 				Assert(str.back() == '"');
 								
 				// Change double-quoted string to single-quoted string
-				str = rewriteDoubleQuotedString(str);		    	
-				rewritten_query_fragment.emplace(std::make_pair(ctx->start->getStartIndex(), std::make_pair(getFullText(ctx->id()), str)));
-			}
+				str = rewriteDoubleQuotedString(str);		    	  	
+				rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair(getFullText(ctx->id()), str)));										
+			}		
 			else if (str.front() == '\'')
 			{
 				// This is a single-quoted string, no further action needed	
 				Assert(str.back() == '\'');		    						
 			}
+			else if (str.front() == '[') 
+			{
+				// When it's a bracketed identifier, remove the delimiters as T-SQL treats it as a string								
+				Assert(str.back() == ']');
+
+				// Temporarily turn this into a double-quoted string so that we can handle embedded quotes.
+				// Since embedded double quotes inside a bracketed identifier are not escaped (as they would be in a 
+				// double-quoted string), escape them first
+				str = escapeDoubleQuotes(str);
+				str = '"' + str.substr(1,str.length()-2) + '"';
+				str = rewriteDoubleQuotedString(str);	
```

Comment:
It looks like `escapeDoubleQuotes()` adds `\"` and `rewriteDoubleQuotedString()` removes `\"`. Could you explain why do we need to add and then remove it? A concrete example would be helpful as well.

Reply:
Please see the comment preceding these lines. An example is `EXECUTE myproc [a"b]`

Reply:
What's going wrong if we do only
```
str = '"' + str.substr(1,str.length()-2) + '"';
str = rewriteDoubleQuotedString(str);
```
for `EXECUTE myproc [a"b]`?

Reply:
When we have two adjacent embedded double quotes, your proposed change would remove one of them. 
I.e.   [a""b]   would then become [a"b] so we'd lose one quote.